server/ntb/io/feeding_services/ntb_reuters_api.py

```diff
@@ -0,0 +1,194 @@
+            self.auth(provider, provider_config)
+
+        headers = {
+            "Authorization": f'Bearer {provider_config.get("token")}',
+            "Content-Type": "application/json",
+        }
+        cursor = ""
+        while True:
+            try:
+                variables = {
+                    "channel": provider_config.get("channel", ""),
+                    "topicCodes": provider_config.get("topic", ""),
+                    "cursor": cursor,
+                }
+                response = self.session.post(
+                    provider_config.get("url"),
+                    headers=headers,
+                    data=json.dumps(
+                        {"query": self.get_query(), "variables": variables}
+                    ),
+                    timeout=30,
+                )
+                response.raise_for_status()
+                data = response.json()
+            except requests.exceptions.HTTPError as e:
+                print(f"HTTP Error: {e}")
+                return
+
+            parser = self.get_feed_parser(provider)
+            items.extend(parser.parse(data, provider))
+
+            val = data.get("data", {}).get("search", {})
+            if val:
+                page_info = val.get("pageInfo")
+                cursor = page_info.get("endCursor")
+                if not page_info.get("hasNextPage"):
+                    break
+            else:
+                break
+
+        if isinstance(items, list):
+            yield items
+        else:
+            yield [items]
+
+    def auth(self, provider, provider_config):
+        provider_config.setdefault(
+            "url", "https://api.reutersconnect.com/content/graphql"
+        )
+        provider_config.setdefault(
+            "auth_url", "https://auth.thomsonreuters.com/oauth/token"
+        )
+
+        # get_Token...
+        auth_url = provider_config.get("auth_url", None)
+        body = {
+            "client_id": provider_config.get("client_id", ""),
+            "client_secret": provider_config.get("client_secret", ""),
+            "grant_type": "client_credentials",
+            "audience": provider_config.get("audience", ""),
+        }
+        response = self.session.post(auth_url, data=body, timeout=30)
+        response.raise_for_status()
+        if response.status_code == 200:
+            data = response.json()
+            provider_config.setdefault("token", data.get("access_token"))
+            superdesk.get_resource_service("ingest_providers").update(
+                provider.get("_id"), self.provider, provider
+            )
+        else:
+            raise IngestApiError.apiAuthError()
+
+    def get_query(self):
+        query = """
+            query MyQuery($channel: [String]!, $topicCodes: [String]!, $cursor: String!) {
+            currentUser {
+                email
+            }
+            search(filter: {channel: $channel, topicCodes: $topicCodes}, cursor: $cursor, limit: 100) {
```

Comment:
imo there should be also filter on `dateRange` so we're not fetching everything but only news since the last update or last 1h on initial load

Reply:
and it would be also useful to be able to use the query param